### DISCUSSION TITLE
is there something wrong with the Charinfo in raygui.h?

### DISCUSSION THREAD

**Discussion by AnasAyadKamal** [raysan5/raylib#1938] [Q&A] [answered]
`/raylib/src/extras/raygui.h:2893:31: error: ‘CharInfo’ undeclared (first use in this function); did you mean ‘BoneInfo’?
                 font.chars = (CharInfo *)RAYGUI_CALLOC(font.charsCount, sizeof(CharInfo));
                               ^~~~~~~~
                               BoneInfo
`
it says change the Charinfo into a Bone info for some odd reason

**Comment by orcmid** [comment, marked as answer]
It's a guess, not a suggestion. 

 The thing to do is find line 2891 in `raygui.h` and tease out what's happening there.  It turns out that `CharInfo` is a struct type that is only defined if `RAYGUI_STANDALONE` is set.  (Found by searching `raygui.h` for `CharInfo` occurrences.)

I suspect that if you are not using `raygui.h` as a standalone (and providing some additional functions yourself), you need to provide `#include "raylib.h"` ahead of any `#include "raygui.h` or `#include "extras/raylib.h"` depending on your setup for includes.

My money is on `CharInfo` being defined in `raylib.h`  and I just lost that bet.  There is no `CharInfo` defined or used in `raylib.h` , although there are a variety of `*Info` thingies, including `BoneInfo`.  It looks like this should be treated as a bug because of that contradiction.  Ramon's comments are concerning :).

**Comment by orcmid** [comment]
I just opened Issue #1939 on this.

**Reply by AnasAyadKamal** [reply]
aha ok

**Reply by orcmid** [reply]
@Boxstop The `raygui.h` has been corrected.  If you are using a clone or a fork of raysan5/raylib, you should synchronize your clone.  If you are using a Zip, get a fresh one :).
